README.md

```diff
@@ -324,6 +324,13 @@ Emitted when a LevelDB query is made using the `leveldown` module.
     * `value` (Object) The value being added to the LevelDB database using the `put` method (Undefined for other methods) 
     * `opCount` (Number) The number of operations carried out by a `batch` method (Undefined for other methods) 
     * `duration` (Number) the time taken for the LevelDB query to be responded to in ms.
+    
+### Event: 'loopback-datasource-juggler'
+Emitted when a function is called on the `loopback-datasource-juggler` module
```

Comment:
I see, it makes more sense now.

I am proposing to reword this sentence as follows:

> Emitted when a `DataAccessObject` function is called. `DataAccessObject` functions are mixed into models that are attached to a database-backed connector, e.g. `MySQL` or `MongoDB`. These functions are exposed to REST clients via LoopBack's `PersistedModel`, you can learn more about them in [LoopBack documentation](https://loopback.io/doc/en/lb3/PersistedModel-REST-API.html).

I am not sure if intercepting a white-list of methods is the best way forward. For example, we have recently added [`upsertWithWhere`](https://github.com/strongloop/loopback-datasource-juggler/blob/b34b6e0ad41e9e2935488ff95c13bf93d892d212/lib/dao.js#L706) that's not included in that list.

IIRC, all methods defined on `DataAccessObject` are mixed into the target model as public API methods, I think it should be safe to intercept them all by simply iterating over all constructor/prototype fields and filtering function values only.

While we are improving this probe, it would be good to intercept `KeyValueAccessObject` too, see https://github.com/strongloop/loopback-datasource-juggler/blob/b34b6e0ad41e9e2935488ff95c13bf93d892d212/lib/kvao/index.js

These two access-objects should cover most paths where LoopBack talks to different databases.

However, there are connectors that don't use any built-in access object: SOAP, REST, Swagger, possibly few more. These connectors build a custom access object depending on the API exposed by the backend service they are connecting to. Installing an intercepter for these methods may be more difficult. However, if you can identify small tweaks we can make in LoopBack to make that task easier, then I think we should include them in the framework.

cc @raymondfeng @kjdelisle